src/components/EventForm/EventForm.tsx

```diff
@@ -0,0 +1,271 @@
+  const [databaseInput, setDatabaseInput] = useState<DefaultForm>({
+    eventName: "",
+    eventCategory: "",
+    eventCapacity: null,
+    eventDescription: "",
+    eventDate: null,
+    eventImages: [""],
+  });
+  const [images, setImages] = useState<string[]>([]);
+  const [inputValue, setInputValue] = useState<string>("");
+  const [required, setRequired] = useState({
+    errorOne: true,
+    errorTwo: true,
+    errorThree: true,
+    errorFour: true,
+    errorFive: true,
+    errorSix: false,
+  });
+  const [submitted, setSubmitted] = useState<boolean>(false);
+
+  const handleEventName = (event: ChangeEvent<HTMLInputElement>) => {
+    const eventName = event.currentTarget.value;
+    setDatabaseInput({ ...databaseInput, eventName: eventName });
+    setRequired({ ...required, errorOne: databaseInput.eventName.length < 1 });
+  };
+  const handleEventCategory = (event: ChangeEvent<HTMLInputElement>) => {
+    const eventCategory = event.currentTarget.value;
+    setDatabaseInput({ ...databaseInput, eventCategory: eventCategory });
+    setRequired({
+      ...required,
+      errorTwo: databaseInput.eventCategory.length < 1,
+    });
+  };
+  const handleEventDate = (event: ChangeEvent<HTMLInputElement>) => {
+    const inputDate = new Date(event.currentTarget.value);
+    const currentDate = new Date();
+    const timestampDate = Timestamp.fromDate(inputDate);
+    setDatabaseInput({
+      ...databaseInput,
+      eventDate: timestampDate,
+    });
+    setRequired({ ...required, errorThree: isBefore(inputDate, currentDate) });
+  };
+  const handleEventCapacity = (event: ChangeEvent<HTMLInputElement>) => {
+    const eventCapacity = event.currentTarget.value;
+    if (
+      !eventCapacity.toLowerCase().search(/[a-z]/g || eventCapacity.length < 1)
+    ) {
+      setDatabaseInput({
+        ...databaseInput,
+        eventCapacity: null,
+      });
+      return;
+    }
+    setDatabaseInput({
+      ...databaseInput,
+      eventCapacity: parseInt(eventCapacity),
+    });
+
+    setRequired({
+      ...required,
+      errorFour: databaseInput.eventCapacity === null,
+    });
+  };
+  const handleEventDescription = (event: ChangeEvent<HTMLTextAreaElement>) => {
+    const eventDescription = event.currentTarget.value;
+    setDatabaseInput({
+      ...databaseInput,
+      eventDescription: eventDescription,
+    });
+
+    setRequired({
+      ...required,
+      errorFive: databaseInput.eventDescription.length < 1,
+    });
+  };
+  const handleAddImage = () => {
+    const image = inputValue;
+    if (!inputValue) {
+      return;
+    }
+    setImages((images) => [...images, image]);
+    setDatabaseInput({ ...databaseInput, eventImages: [...images, image] });
+    setInputValue("");
+
+    setRequired({
+      ...required,
+      errorSix: databaseInput.eventImages.length === 0,
+    });
+  };
+  const handleRemoveImage = () => {
+    if (images.length > 0) {
+      images.pop();
```

Comment:
Using `.pop()` will unintentionally update your state which can lead to unexpected problems as you are not using the update state function. 

It could be something like this.

```ts
const updatedImages = [...images]
if (updatedImages .length > 0) {
      updatedImages .pop();
   }
   setImages(updatedImages);

```